### ISSUE TITLE
Question: Minver failing builds in Rider, but works in CLI

### ISSUE THREAD

**Issue by rogeralsing** [adamralph/minver#429]
I'm stuck on this issue where Minver fails my builds in Rider, building in CLI works fine.
Any pointers on where I should start looking for the cause?

Removing minver deps and from manifest makes it possible to compile again.

<img width="1722" alt="Skärmavbild 2020-12-05 kl  11 27 27" src="https://user-images.githubusercontent.com/647031/101240034-e198c780-36ec-11eb-8dee-566107e62459.png">

<img width="791" alt="Skärmavbild 2020-12-05 kl  10 58 49" src="https://user-images.githubusercontent.com/647031/101240007-97174b00-36ec-11eb-85fd-be38c6e7b088.png">

**Comment by rogeralsing** [comment]
More context, it fails on the version override. where are these args coming from? 

<img width="1259" alt="Skärmavbild 2020-12-05 kl  11 37 20" src="https://user-images.githubusercontent.com/647031/101240256-5caead80-36ee-11eb-8805-ace4565cb605.png">

**Comment by rogeralsing** [comment]
log output

```
CONSOLE: Build started 12/05/2020 12:25:26.
0>------- Started building project: Proto.Actor
1>------- Started building project: Proto.Persistence
2>------- Started building project: Messages
3>------- Started building project: ProtoGrainGenerator
0>dotnet "/Users/rogerjohansson/.nuget/packages/minver/2.3.1/build/../minver/MinVer.dll" --auto-increment "" --build-metadata "" --default-pre-release-phase "" --minimum-major-minor "" --repo "/Users/rogerjohansson/git/asynkron/protoactor-dotnet/src/Proto.Actor" --tag-prefix "" --verbosity ""
0>/var/folders/bt/8r9hr2r96md84wzv_t7h0kdh0000gn/T/tmpc65e28713bf4468aab8fb91e18c9473b.exec.cmd: line 2: dotnet: command not found
0>/Users/rogerjohansson/.nuget/packages/minver/2.3.1/build/MinVer.targets(39,5): Error MSB3073 : The command "dotnet "/Users/rogerjohansson/.nuget/packages/minver/2.3.1/build/../minver/MinVer.dll" --auto-increment "" --build-metadata "" --default-pre-release-phase "" --minimum-major-minor "" --repo "/Users/rogerjohansson/git/asynkron/protoactor-dotnet/src/Proto.Actor" --tag-prefix "" --verbosity """ exited with code 127.
```

**Comment by rogeralsing** [comment]
Found the source of the issue, not sure why it is though.
Starting `tmux` in zsh.rc, breaks this for me.

removing tmux makes everything work normally again.. live and learn..

**Comment by adamralph** [comment]
> Starting `tmux` in zsh.rc, breaks this for me.
> 
> removing tmux makes everything work normally again.. live and learn.. 

I've heard some of those words before, but not all of them. 🙂 

> it fails on the version override

FWIW, it looks like the wrong version of the minver executable was being invoked by the .NET SDK target contained in the `MinVer` package. Perhaps it was invoking an older version of `minver-cli` installed globally on your system.

**Comment by luizhlelis** [comment]
Hello @adamralph and @rogeralsing.
I'm facing exactly the same error when building a project with `MinVer` 2.3.0. As @rogeralsing, I'm also running it on MacOs with [zsh](https://github.com/ohmyzsh/ohmyzsh) but I'm not using `tmux`. In my case, the difference is: both Ryder and CLI are throwing that error. `MinVer` is running the command bellow:

```bash
dotnet "<NUGET-PACKAGE-PATH>/minver/2.3.0/build/../minver/MinVer.dll" --auto-increment "" --build-metadata "" --default-pre-release-phase "" --minimum-major-minor "" --repo "<PROJECT-PATH>" --tag-prefix "core-" --verbosity "detailed" --version-override "" 
```

I tried to run that command in terminal and it's throwing the error bellow:

```bash
Failed to initialize CoreCLR, HRESULT: 0x80004005
```

I've already tried some workaround (based in some references: [stackoverflow](https://stackoverflow.com/questions/60232802/unable-to-build-identityserver4-on-windows-10-machine), [gh issue](https://github.com/adamralph/minver/issues/318)) without success:

- Install `minver-cli`;
- Reinstall `dotnet` sdk;
- Reinstall `git`;

PS: I tried to build it with docker with a `debian` image and everything worked fine.

**Comment by adamralph** [comment]
I'm afraid I don't have macOS, so I can't reproduce the problem.

Perhaps you can try and build your own hello world console app and run it in a similar way? E.g. `dotnet helloworld.dll`.

**Comment by rogeralsing** [comment]
My impression is that the problem is different paths to dotnet depending on environment.
e.g in my case, there were some env path to dotnet for my default console, and another when starting TMUX.

I never really looked into this more, I just stopped using TMUX :-P

**Comment by luizhlelis** [comment]
Hello @rogeralsing and @adamralph, thanks for your quickly response.
I've just solved the error. Inside `$HOME/.nuget/packages/minver/2.3.0/minver/` folder there is a `MinVer.runtimeconfig.json` file which has the following config:

```json
{
  "runtimeOptions": {
    "tfm": "netcoreapp2.1",
    "rollForward": "major",
    "framework": {
      "name": "Microsoft.NETCore.App",
      "version": "2.1.0"
    }
  }
}
```

and I realized that I don't have `netcoreapp2.1` installed locally (I only have the newest ones). After installing the `dotnet sdk 2.1.818`, I was able to build my project and no error was thrown.
